### PR TITLE
Nj 198 duplicate pdf field - followup to #5155

### PR DESCRIPTION
**Pull request by mluedke2**
## Link to pivotal/JIRA issue

https://github.com/newjersey/affordability-pm/issues/198

## Is PM acceptance required?
- No - merge after code review approval

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- I used [PDFEscape](https://www.pdfescape.com/) to remove one of the duplicated fields and replace it with a new field with the name `Text15`.
- For some reason, exporting from PDFEscape reverted all those changes from the diff of #5155 ... my sincere apologies for having another big diff that just reverts all those. Had I known that would happen, I would have done this in the previous PR and that diff would have been small indeed.
- I considered abstracting `nil` and `""` to be equivalent so that the diff would be small... but I'm going to put my chips on the side of the PDF not needing replacement often. If anyone feels differently, I wouldn't mind doing that at all.

## How to test?
- Open up the new PDF and insert separate values in the formerly-entangled boxes (3rd box in line 42 and the 2nd box in line 55)

## Screenshots (for visual changes)

Behold, separate values! (Yes I know the gray box looks different -- I ran it past Product and got the ok)
image

### CODE REVIEW

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb [hunk 1]
```diff
@@ -989,20 +989,6 @@
...
-            "Choice1"
-          when "1"
-            "Off"
-          when nil
-            ""
-          else
-            input
-          end
-        end
-
```
Comment (mluedke2) [inline spec/lib/pdf_filler/nj1040_pdf_spec.rb:1005]: goodbye!

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb [hunk 2]
```diff
@@ -1464,6 +1446,25 @@ def map_spouse_death_year(input)
...
+        expect(pdf_fields["Text20"]).to eq ""
+        expect(pdf_fields["Text30"]).to eq ""
+        # hundreds
+        expect(pdf_fields["Text37"]).to eq ""
+        expect(pdf_fields["Text38"]).to eq ""
+        expect(pdf_fields["Text39"]).to eq ""
+        # decimals
+        expect(pdf_fields["Text40"]).to eq ""
+        expect(pdf_fields["Text41"]).to eq ""
+      end
```
Comment (mluedke2) [inline spec/lib/pdf_filler/nj1040_pdf_spec.rb:1467]: i added one test here because there wasn't yet one for the case where this is blank

**Review by jnf — APPROVED** [review]

**Review by jachan — APPROVED** [review]
Looks good!

File: app/lib/pdf_filler/nj1040_pdf.rb [hunk 3]
```diff
@@ -470,7 +470,7 @@ def hash_for_pdf
           "undefined_145",
           "undefined_144",
           "undefined_143",
-          "undefined_114",
+          "Text15",
```
[the remark this answers is not in the document]
Comment (mluedke2) [reply app/lib/pdf_filler/nj1040_pdf.rb:473]: great idea, will add a comment...

[the remark this answers is not in the document]
Comment (mluedke2) [reply app/lib/pdf_filler/nj1040_pdf.rb]: done 👍

**Pull request merged**